src/modules/75-cf/pages/environment-details/__tests__/EnvironmentDetails.test.tsx

```diff
@@ -64,71 +92,82 @@ describe('EnvironmentDetails', () => {
       useSyncedEnvironment: () => ({ data: mockEnvironment, loading: undefined, error: undefined, refetch: jest.fn() })
     })
 
-    render(
-      <TestWrapper
-        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <EnvironmentDetails />
-      </TestWrapper>
-    )
+    renderComponent()
 
     expect(getAllByText(document.body, mockEnvironment.data.name)).toBeDefined()
     expect(getAllByText(document.body, mockEnvironment.data.identifier)).toBeDefined()
 
     expect(getAllByText(document.body, mockApiKeys.apiKeys[0].name)).toBeDefined()
-    expect(getAllByText(document.body, mockApiKeys.apiKeys[1].name)).toBeDefined()
   })
 
-  test('CFEnvironmentDetailsBody should render loading', async () => {
+  test('should render toolbar when there are existing SDK Keys ', async () => {
+    mockImport('services/cd-ng', {
+      useGetEnvironment: () => ({ data: mockEnvironment, loading: false, error: undefined, refetch: jest.fn() })
+    })
+
     mockImport('services/cf', {
       useGetAllAPIKeys: () => ({ data: mockApiKeys })
     })
 
-    render(
-      <TestWrapper
-        path="/account/:accountId/cf/orgs/:orgIdentifier/projects/:projectIdentifier/onboarding/detail"
-        pathParams={{ accountId: 'dummy', orgIdentifier: 'dummy', projectIdentifier: 'dummy' }}
-      >
-        <EnvironmentDetailsBody environment={mockEnvironment.data as EnvironmentResponseDTO} />
-      </TestWrapper>
-    )
+    renderComponent()
 
-    expect(document.body.querySelector('[class*=ContainerSpinner]')).toBeDefined()
+    await waitFor(() => {
+      expect(screen.getByRole('button', { name: 'cf.environments.apiKeys.addKeyTitle' })).toBeInTheDocument()
+      expect(document.getElementsByClassName('PageSubHeader--container')[0]).toBeInTheDocument()
```

Comment:
An easier way than `getElementsByClassName` is with `querySelector`. Ideally assert that a specific thing (button, text etc) is available.
```suggestion
      expect(document.querySelector('.PageSubHeader--container')).toBeInTheDocument()
```

Reply:
updated.
I've asserted the text button appears too